src/lib/operators/join_hash/join_hash_steps.hpp

```diff
@@ -406,98 +402,114 @@ void probe(const RadixContainer<RightType>& radix_container,
         pos_list_right_local.reserve(static_cast<size_t>(expected_output_size));
 
         for (size_t partition_offset = partition_begin; partition_offset < partition_end; ++partition_offset) {
-          auto& row = partition[partition_offset];
+          auto& right_row = partition[partition_offset];
 
-          if (mode == JoinMode::Inner && row.row_id == NULL_ROW_ID) {
+          if (mode == JoinMode::Inner && right_row.row_id == NULL_ROW_ID) {
             // From previous joins, we could potentially have NULL values that do not refer to
-            // an actual row but to the NULL_ROW_ID. Hence, we can only skip for inner joins.
+            // an actual right_row but to the NULL_ROW_ID. Hence, we can only skip for inner joins.
             continue;
           }
 
-          const auto& rows_iter = hashtable.find(type_cast<HashedType>(row.value));
+          const auto& rows_iter = hash_table.find(type_cast<HashedType>(right_row.value));
 
-          if (rows_iter != hashtable.end()) {
-            // Key exists, thus we have at least one hit
-            const auto& matching_rows = rows_iter->second;
+          if (rows_iter != hash_table.end()) {
+            // Key exists, thus we have at least one hit for the primary predicate
+            const auto& primary_predicate_matching_rows = rows_iter->second;
 
             // Since we cannot store NULL values directly in off-the-shelf containers,
             // we need to the check the NULL bit vector here because a NULL value (represented
             // as a zero) yields the same rows as an actual zero value.
             // For inner joins, we skip NULL values and output them for outer joins.
-            // Note, if the materialization/radix partitioning phase did not explicitely consider
+            // Note, if the materialization/radix partitioning phase did not explicitly consider
             // NULL values, they will not be handed to the probe function.
-            if constexpr (consider_null_values) {
+            if constexpr (retain_null_values) {
               if ((*radix_container.null_value_bitvector)[partition_offset]) {
-                if (mode == JoinMode::Left || mode == JoinMode::Right) {
-                  pos_list_left_local.emplace_back(NULL_ROW_ID);
-                  pos_list_right_local.emplace_back(row.row_id);
-                }
+                pos_list_left_local.emplace_back(NULL_ROW_ID);
```

Comment:
Why don't we consider secondary predicates here?

Reply:
If the primary predicate is NULL, we do not need to care about the others